### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by hackallcode**
Changes:
- use AWS instead of download.tarantool.io, because old TDG versions are now only available there;
- update version of TNT bundle, TDG;
- remove old CA certificate in Centos docker image;
- create caches of python dependencies;
- use pip2 instead of pip3 with python 2;
- specify Python dependencies for Python 2;
- fix CI matrix for Python 2;
- fix unit tests for Python 2;
- fix molecule tests for Python 2;
- update the last version of Ansible to 4.8.0 in CI;
- remove 2.10.0 Ansible in CI (it doesn't work with Molecule package on Python 2 and Python 3).

### CODE REVIEW

File: .github/helpers/count_molecule_matrix.py [hunk 1]
```diff
@@ -22,23 +22,24 @@
 
 DEFAULT_ANSIBLE_VERSION = '2.8.0'
 NOT_DEFAULT_ANSIBLE_VERSIONS = [
-    '2.9.0',
-    '2.10.0',
-    '4.1.0',
+    '2.9.27',
+    '4.8.0',
```
Comment (dokshina) [inline .github/helpers/count_molecule_matrix.py:26]: This change isn't described in the description, so I don't understand why these versions are tested now.

File: .github/helpers/count_molecule_matrix.py [hunk 2]
```diff
@@ -58,39 +59,63 @@
...
-    return {
-        **get_matrix_base(molecule_scenario, ansible_version),
-        'tdg_version': tdg_version or DEFAULT_TDG_VERSION,
-        'molecule_command': get_molecule_command(molecule_command),
-    }
+def get_ce_params(
+    molecule_scenario=None, ansible_version=None, python_version=None,
+    tarantool_version=None, molecule_command=None,
+):
+    matrix = get_matrix_base(molecule_scenario, ansible_version, python_version)
```
Comment (dokshina) [inline .github/helpers/count_molecule_matrix.py:89]: Why did you change the way these dictionaries are created?

File: .github/helpers/count_molecule_matrix.py [hunk 3]
```diff
@@ -113,7 +138,10 @@ def main(event_name, repo_owner, review_state, ref):
             ce_matrix.append(get_ce_params(molecule_command=command))
 
         for version in NOT_DEFAULT_ANSIBLE_VERSIONS:
-            ce_matrix.append(get_ce_params(ansible_version=version))
+            python_version = None
+            if compare_versions(version, '2.10.0') >= 0:
```
Comment (dokshina) [inline .github/helpers/count_molecule_matrix.py]: There are 2 "version" variables here, let's rename `version -> ansible_version`.

File: .github/helpers/count_molecule_matrix.py [hunk 4]
```diff
@@ -125,13 +153,13 @@ def main(event_name, repo_owner, review_state, ref):
...
+    print('::set-output name=ce-tests-found::' + "true" if len(ce_matrix) > 0 else "false")
+    print('::set-output name=ee-tests-found::' + "true" if len(ee_matrix) > 0 else "false")
+    print('::set-output name=tdg-tests-found::' + "true" if len(tdg_matrix) > 0 else "false")
 
-    print(f'::set-output name=ce-matrix::{json.dumps({"include": ce_matrix})}')
-    print(f'::set-output name=ee-matrix::{json.dumps({"include": ee_matrix})}')
-    print(f'::set-output name=tdg-matrix::{json.dumps({"include": tdg_matrix})}')
+    print('::set-output name=ce-matrix::' + json.dumps({"include": ce_matrix}))
+    print('::set-output name=ee-matrix::' + json.dumps({"include": ee_matrix}))
+    print('::set-output name=tdg-matrix::' + json.dumps({"include": tdg_matrix}))
```
Comment (dokshina) [inline .github/helpers/count_molecule_matrix.py:162]: This is because of using Python2?

File: .github/workflows/molecule-tests.yml [hunk 5]
```diff
@@ -14,11 +14,12 @@ jobs:
       - name: Setup python
         uses: actions/setup-python@v2
         with:
-          python-version: '2.7'
+          python-version: '3.9'
```
Comment (dokshina) [inline .github/workflows/molecule-tests.yml:17]: But here Python 3 is used for generating matrix.

File: .github/workflows/molecule-tests.yml [hunk 6]
```diff
@@ -44,13 +45,28 @@ jobs:
...
         run: |
-          pip3 install --upgrade -r requirements.txt
-          pip3 uninstall -y ansible ansible-base ansible-core
-          pip3 install ansible~=${{ matrix.ansible_version }}
+          python -V
+          if [ "${{ matrix.python_version }}" = "2.7" ]; then
+              pip install virtualenv
+              virtualenv venv
+          else
+              python -m venv venv
```
Comment (dokshina) [inline .github/workflows/molecule-tests.yml:65]: BTW, did you check GH Actions modules for python venv?

File: .github/workflows/unit-tests.yml [hunk 7]
```diff
@@ -11,38 +11,56 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        tarantool-version: ["1.10"]
+        python_version: ["2.7", "3.9"]
```
Comment (dokshina) [inline .github/workflows/unit-tests.yml:14]: Why do you use `_` here, but `-` above?

File: molecule/rolling_update/converge.yml [hunk 8]
```diff
@@ -13,7 +13,7 @@
     cartridge_scenario:
       - deliver_package
       - update_package
-    cartridge_package_path: ./packages/myapp-5.0.0-0-with-c-2.5.0.tar.gz
+    cartridge_package_path: ./packages/myapp-5.0.0-0-with-c-2.6.0.tar.gz
```
Comment (dokshina) [inline molecule/rolling_update/converge.yml:16]: I can be wrong, but the reason for testing special versions was the difference in the internal Cartridge API.

File: molecule/rolling_update/tests/test_dists_rotated.py [hunk 9]
```diff
@@ -1,10 +1,24 @@
...
+
+from testinfra.modules.file import File
 
 import utils
 
 testinfra_hosts = utils.get_testinfra_hosts()
 
 
+def listdir(self):
+    out = self.run_test("ls -1 -q -- %s", self.path)
```
Comment (dokshina) [inline molecule/rolling_update/tests/test_dists_rotated.py:12]: Wasn't mentioned in description, I didn't understand.

**Review by dokshina — COMMENTED** [review]
IMHO, there are enough changes to split this PR into 3 or 2.

File: .github/helpers/count_molecule_matrix.py (hunk 2 above)
Comment (hackallcode) [reply .github/helpers/count_molecule_matrix.py:89]: Because spread operator is not supported by Python 2

File: .github/helpers/count_molecule_matrix.py (hunk 1 above)
Comment (hackallcode) [reply .github/helpers/count_molecule_matrix.py:26]: Added:
```
- update the last version of Ansible to 4.8.0 in CI;
- remove 2.10.0 Ansible in CI (it doesn't work with Molecule package on Python 2 and Python 3);
```

File: .github/workflows/unit-tests.yml (hunk 7 above)
Comment (hackallcode) [reply .github/workflows/unit-tests.yml:14]: Where? `_` is used everywhere

File: molecule/rolling_update/converge.yml (hunk 8 above)
Comment (hackallcode) [reply molecule/rolling_update/converge.yml:16]: 2.4.0 doesn't work :)

More info here:
https://github.com/tarantool/sdk/issues/245

But it used only to update from the old version to the new version, so 2.5.0 + 2.6.0 is good too

File: molecule/rolling_update/tests/test_dists_rotated.py (hunk 9 above)
Comment (hackallcode) [reply molecule/rolling_update/tests/test_dists_rotated.py:12]: This function doesn't exists in Python 2, so it's a copy-paste from Python 3 library

File: .github/workflows/molecule-tests.yml (hunk 6 above)
Comment (hackallcode) [reply .github/workflows/molecule-tests.yml:65]: What GH Actions modules do you mean? Actions? In any case, importing action is harder than this if

File: .github/workflows/molecule-tests.yml (hunk 5 above)
Comment (hackallcode) [reply .github/workflows/molecule-tests.yml:17]: Linter flake8 is run by Python 2 and Python 3. It checks this file too. So, this file should be compatible with both versions of Python.

File: .github/helpers/count_molecule_matrix.py (hunk 4 above)
Comment (hackallcode) [reply .github/helpers/count_molecule_matrix.py:162]: Yes

File: .github/helpers/count_molecule_matrix.py (hunk 3 above)
Comment (hackallcode) [reply .github/helpers/count_molecule_matrix.py]: Fixed

**Review by dokshina — APPROVED** [review]
